src/module-loader.js

```diff
@@ -92,12 +96,36 @@ export class ModuleLoader extends EventTarget {
         nodePath.join(this.basePath, directory, file.name)
       );
 
-      this.registry.add(
-        `/${nodePath.join(directory, nodePath.parse(file.name).name)}`,
-        endpoint
-      );
+      if (file.name.includes("$context")) {
+        this.addContext(directory, endpoint);
+      } else {
+        this.registry.add(
+          `/${nodePath.join(directory, nodePath.parse(file.name).name)}`,
+          endpoint
+        );
+      }
     });
 
     await Promise.all(imports);
   }
+
+  addContext(directory, endpoint) {
+    this.contextRegistry.add(`/${directory}`, endpoint.default);
+  }
+
+  updateContext(directory, endpoint) {
+    const context = this.contextRegistry.find(directory);
+    const updatedContext = endpoint.default;
+
+    for (const property in updatedContext) {
+      if (
+        Object.prototype.hasOwnProperty.call(updatedContext, property) &&
+        !Object.prototype.hasOwnProperty.call(context, property)
+      ) {
+        context[property] = updatedContext[property];
+      }
+    }
+
+    Object.setPrototypeOf(context, Object.getPrototypeOf(updatedContext));
+  }
```

Comment:
- [ ] move `updateContext()` logic into `ContextRegistry`
- [ ] unit tests for `contextRegistry.update()`